### PR TITLE
_Ty -> _TVal for atomic_ref, one missed occurrence

### LINKED ISSUE #1302: <atomic>: atomic_ref<int8_t>::operator--(int) fails to compile
There is an incorrect cast on line 1368 of &lt;atomic&gt;. The cast to _Ty should be _TVal. 

    _TVal operator--(int) noexcept {
        return static_cast<_Ty>(_InterlockedExchangeAdd8(_Atomic_address_as<char>(this->_Storage), -1));
    }

**Command-line test case**
```
C:\Temp>type repro.cpp
#include <atomic>

int main() {
	int8_t x = 0;
	std::atomic_ref(x)--;
}

C:\Temp>cl /c /std:c++latest .\repro.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29304.1 for x86
Copyright (C) Microsoft Corporation.  All rights reserved.

/std:c++latest is provided as a preview of language features from the latest C++
working draft, and we're eager to hear about bugs and suggestions for improvements.
However, note that these features are provided as-is without support, and subject
to changes or removal as the working draft evolves. See
https://go.microsoft.com/fwlink/?linkid=2045807 for details.

repro.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29304\include\atomic(1368): error C2440: 'static_cast': cannot convert from 'char' to '_Ty'
        with
        [
            _Ty=int8_t &
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29304\include\atomic(1368): note: static_cast and safe_cast to reference can only be used for valid initializations or for lvalue casts between related classes
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29304\include\atomic(1367): note: while compiling class template member function 'signed char std::_Atomic_integral<_Ty &,1>::operator --(int) noexcept'
        with
        [
            _Ty=int8_t
        ]
repro.cpp(5): note: see reference to function template instantiation 'signed char std::_Atomic_integral<_Ty &,1>::operator --(int) noexcept' being compiled
        with
        [
            _Ty=int8_t
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29304\include\atomic(1781): note: see reference to class template instantiation 'std::_Atomic_integral<_Ty &,1>' being compiled
        with
        [
            _Ty=int8_t
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29304\include\atomic(2328): note: see reference to class template instantiation 'std::_Atomic_integral_facade<_Ty &>' being compiled
        with
        [
            _Ty=int8_t
        ]
repro.cpp(5): note: see reference to class template instantiation 'std::atomic_ref<int8_t>' being compiled
```

**Expected behavior**
Should compile successfully.

**STL version**
Latest

### PR DESCRIPTION
**Pull request by AlexGuteniev**
Resolves #1302

### CODE REVIEW

**Review by miscco — CHANGES_REQUESTED** [review]
For what its worth, this looks correct. 

I would prefer that we add a test though

**Review by miscco — APPROVED** [review]

**Comment by miscco** [comment]
🎉 for increased test coverage

File: tests/std/tests/P0019R8_atomic_ref/test.cpp [hunk 1]
```diff
@@ -133,6 +133,14 @@ void test_int_ops() {
     assert(vy.load() == 0x40);
     assert(rx.load() == 0x33);
     assert(ry.load() == 0x33);
+
+    vx--;
+    rx--;
```
Comment (CaseyCarter) [inline tests/std/tests/P0019R8_atomic_ref/test.cpp:138]: We assert that all of these read-modify-write operations change the stored value correctly, but not that they yield the correct result. Should we?

**Review by CaseyCarter — APPROVED** [review]

File: tests/std/tests/P0019R8_atomic_ref/test.cpp (hunk 1 above)
Comment (AlexGuteniev) [reply tests/std/tests/P0019R8_atomic_ref/test.cpp:138]: yes, added

**Review by CaseyCarter — APPROVED** [review]
Awesome - thanks!

**Review by StephanTLavavej — APPROVED** [review]
Thanks for the fix and comprehensive test coverage! :smile_cat:

**Comment by StephanTLavavej** [comment]
Thanks again for the quick fix! :rocket:

**Pull request merged**